### ISSUE TITLE
[Feature Request]: Defer recommended payment until once installed

### ISSUE THREAD

**Issue by pjlsergeant** [jordanbaird/Ice#73]
### Description

I think you're well on your way to becoming a replacement for Bartender, so, congrats.

I think you'll get paid more for your efforts if you find a way to defer the request for payment until after a user is setup and it's working. Unwilling to pay for an app I haven't used before at download time, because I've no idea if it will work, but a few days into it being useful and installed, I'm absolutely willing to pay. But having downloaded it for free, it's now pretty unintuitive how to give you some $$.

I can find the Support link in About, although I don't especially want to set up a recurring donation, so actually I have to go back to the website, click Download again, and fill out the payment again, which I've now done, although presumably I've also screwed up any analytics you have on how much people are willing to pay by creating both a $0 and a $25 entry

I think it would be worth considering selling an optional nagware license instead so you're asking people for money a few days after they've used it.

Anyway, great work.

**Comment by alkusin** [comment]
> I can find the Support link in About, although I don't especially want to set up a recurring donation

You can choose "one time" instead of monthly ("Select a tier" above amount textbox).

But a link to a more universal donation system could be better, especially for people who don't have a GitHub account. Or, yeah, a time-limited free trial. I guess Jordan didn't bother overthinking it while crafting Ice in his corner. Today it's different.

**Comment by CrypticComputation** [comment]
Adding my two cents to this as I agree that a good developers time is indeed worth the money. 

I use another open source app here on GitHub that allows self compilation in Xcode (self-compiled) to continue being used for free (the app understands that it was compiled)

It then shows the option to donate if so inclined upon launching, instead of buying

While the standard download is a "demo" giving the user an X amount of days trial

It seems like the best of both worlds for both developers and open source

This is all obviously down the road, as I'm quite sure @jordanbaird wasn't expecting such a quick blow up overnight, but it is indeed an opportunity for an indie developer over a disingenuous organization who fails at transparency

I would happily pay the 16 Bartender was asking for once Ice has matured

**Comment by jordanbaird** [comment]
Thanks for bringing this up. I agree, it's not very intuitive currently. I've also thought about making an "Ice Pro" that contains the full featureset, while keeping the open source version completely free, with roughly the current featureset. Does this seem like a good idea?

**Comment by CrypticComputation** [comment]
Seems that may be the route to go. More-so considering the folks today who have said they don't care about menu bar styling or the permissions it requires. Styling and further manipulation of the menu bar seem worth being in a more premium version later on

**Comment by alkusin** [comment]
Yeah I agree, I don't care about styling and other stuff, especially since (if I understood well) it is why screen recording is necessary. Free version allowing hiding icons and trigger conditions, without the always hidden bug, is perfectly fine to me. Basic, but more polished than the competitors.

Already said in another bug/feature issue, thank you and keep up the good work Jordan, Ice is going to be the best Bartender alternative.

**Comment by SputNikPlop** [comment]
What if instead you offered a precompiled version which would be totally free then there could be certain features you unlock with a license (possible trial as well), it also would be possible to still keep this open source for anyone to develop if they choose to compile from the source. Depending on how much time I have I might email you or whip something up.

**Comment by jordanbaird** [comment]
Update on this: As of right now, I think I'll keep just one version of Ice and keep it free. A possible future direction once Ice is more mature would be to rewrite the app and release Ice 2, which would be a paid upgrade. But that's a ways in the future. As of now I have multiple donation options available, so if there are users who want to support, they aren't required to use Gumroad. What are everyone's thoughts? Would it still be good to have some sort of notification in the app asking for donations, or would that be too intrusive?

**Comment by EthanWonCho** [comment]
What I thought "intrusive" was notifications popping up when some event happens(e.g. opening the app, clicking the certain menu)
I suggest to add notifications every 1 week, so the user would know how to support Ice.

**Comment by pjlsergeant** [comment]
<img width="463" alt="image" src="https://github.com/user-attachments/assets/1607496e-4c83-41fe-b97b-40f03fd7b8dc">

popping up something like this every 7 days seems pretty reasonable
